File: Csharp/Behavioral/Mediator/MR1/Mediator/BaseComponent.cs
namespace TheRayCode.Mediator
{
    class BaseComponent
    {
        protected Mediator _mediator;

        public BaseComponent(Mediator mediator = null)
        {
            this._mediator = mediator;
        }

        public void SetMediator(Mediator mediator)
        {
            this._mediator = mediator;
        }
    }

}

File: Language/PHP/Behavioral/Mediator/Example/BaseComponent.php
<?php


class BaseComponent
{
    protected $mediator;

    public function __construct(Mediator $mediator = null)
    {
        $this->mediator = $mediator;
    }

    public function setMediator(Mediator $mediator): void
    {
        $this->mediator = $mediator;
    }
}